src/platform/plugins/shared/data/public/search/search_interceptor/search_interceptor.ts

```diff
@@ -398,6 +403,10 @@ export class SearchInterceptor {
       catchError((e: Error) => {
         // If we aborted (search:timeout advanced setting) and there was a partial response, return it instead of just erroring out
         if (searchAbortController.isTimeout()) {
+          this.startRenderServices.analytics.reportEvent(EVENT_TYPE_DATA_SEARCH_TIMEOUT, {
+            [EVENT_PROPERTY_SEARCH_TIMEOUT_MS]: this.searchTimeout,
+            [EVENT_PROPERTY_EXECUTION_CONTEXT]: options.executionContext,
+          });
```

Comment:
It might be worth adding a simple unit test for this in case of future refactorings.

Reply:
e0fc53b790e30146493231efcdc17022ffaa79c2